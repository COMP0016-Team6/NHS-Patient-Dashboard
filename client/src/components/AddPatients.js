import React from "react";
import { Link } from "react-router-dom";
import { submitAddPatients } from "../api/fetches";
import { useSelector } from "react-redux";
import { toast } from "react-toastify";
import SearchBar from "./SearchBar";
import { Grid } from '@material-ui/core';
import { useStylesUser } from "../styles/styles";
import NavBar from "./Navbar";

const AddPatients = ({ logout }) => {
  const classes = useStylesUser();
  const myPatients = useSelector(state => state.patients);

  const onSubmitForm = async e => {
    e.preventDefault();
    try {
      const parseRes = await submitAddPatients(myPatients);
      if (parseRes.result === "success") {
        toast.success("Patients are successfully added!");
      } else {
        toast.success(parseRes);
      }
    } catch (err) {
        console.error(err.message);
    }
  };

  return (
    <div className={classes.root}>
      <NavBar heading="" logout={logout} />

      <Grid container justify="center" alignItems="center">
        <Grid item md={6}>
          <Link to="/dashboard">
            <button className="btn btn-primary mt-5">Back</button>
          </Link>
          <h1 className="text-center">Add Patients</h1>
        </Grid>
      </Grid>
    
      <form className="mt-3" onSubmit={onSubmitForm}>

      <Grid container direction="row" justify="center" alignItems="center">
        <Grid item md={6}> <SearchBar select={true}/></Grid>
      </Grid>
      <Grid container direction="row" justify="center" alignItems="center">
        <Grid md={6}>
          <button className="btn btn-success btn-block mt-5">Submit</button>
        </Grid>
      </Grid> 
      </form>
    </div>
  )
}

export default AddPatients;