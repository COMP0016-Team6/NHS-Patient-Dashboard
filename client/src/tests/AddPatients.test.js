import React from "react";
import AddPatients from "../components/AddPatients";
import renderer from "react-test-renderer";
import { wrapProvider } from "./util";
import { BrowserRouter } from 'react-router-dom';

it ("AddPatients snapshot matches", () => {
  const tree = renderer.create(wrapProvider(<BrowserRouter><AddPatients /></BrowserRouter>));
  expect(tree).toMatchSnapshot();
});
