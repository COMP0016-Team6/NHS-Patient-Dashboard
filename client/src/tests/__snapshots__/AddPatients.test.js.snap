// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPatients snapshot matches 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    style={
      Object {
        "background": "#066bdd",
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h4
        className="makeStyles-title-2"
      >
        
         
      </h4>
      <button
        className="btn btn-info"
      >
        Logout
      </button>
    </div>
  </header>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
    >
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary mt-5"
        >
          Back
        </button>
      </a>
      <h1
        className="text-center"
      >
        Add Patients
      </h1>
    </div>
  </div>
  <form
    className="mt-3"
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
      >
         
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
      >
        <button
          className="btn btn-success btn-block mt-5"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
